unison-cli/src/Unison/LSP/Completion.hs

```diff
@@ -240,3 +252,73 @@ matchCompletions (CompletionTree tree) txt =
       let childMatches = mkDefMatches rest <&> over _1 (Path.cons ns)
       let currentMatches = matches <&> \(name, dep) -> (Path.singleton ns, name, dep)
       currentMatches <> childMatches
+
+-- | Called to resolve additional details for a completion item that the user is considering.
+completionItemResolveHandler :: RequestMessage 'CompletionItemResolve -> (Either ResponseError CompletionItem -> Lsp ()) -> Lsp ()
+completionItemResolveHandler message respond = do
+  let completion :: CompletionItem
+      completion = message ^. params
+  respond . maybe (Right completion) Right =<< runMaybeT do
```

Comment:
Why do we return `Right completion` on a parse error rather than some `ResponseError`?

Reply:
I'm not exactly sure how clients handle an error on completion resolve, but my reasoning is that just because we can't add the type signature doesn't mean we should fail entirely, the completion item is still useful to the client and still contains everything except the type signature.